PatternPal/PatternPal.Core/Checks/ParameterCheck.cs

```diff
@@ -16,6 +28,104 @@ internal class ParameterCheck : CheckBase
+            };
+        }
+
+        // For each typecheck, check whether one of the parameters of method has the correct type
+        // and if so remove from method parameters list.
+        foreach (TypeCheck typecheck in _parameterTypes)
+        {
+            bool noneCorrect = true;
+
+            // There are no parameters to check with.   
+            if (methodParameters.Count == 0)
+            {
+                // TODO
+                ICheckResult temp = new LeafCheckResult
+                {
+                    Priority = Priority,
+                    Correct = false,
+                    FeedbackMessage = "There are less parameters than TypeChecks"
+                };
+                subCheckResultsResults.Add(temp);
+                break;
+            }
+
+            for (int x = 0; x < methodParameters.Count(); x++)
+            {
+                var test = methodParameters[x];
+                ICheckResult tempCheck;
+                tempCheck = typecheck.Check(ctx, methodParameters.ElementAt(x));
+
+                switch (tempCheck)
+                {
+                    case LeafCheckResult leafCheckResult:
+                    {
+                        if (leafCheckResult.Correct)
+                        {
+                            subCheckResultsResults.Add(tempCheck);
+                            methodParameters.RemoveAt(x);
+                            noneCorrect = false;
+                            break;
+                        }
+                        continue;
+                    }
+                    case NodeCheckResult nodeCheckResult:
+                    {
+                        bool anyFalse = false;
+                        foreach (LeafCheckResult subcheck in nodeCheckResult.ChildrenCheckResults)
+                        {
+                            // One of the subchecks was incorrect.
+                            if (!subcheck.Correct)
+                            {
+                                anyFalse = true;
+                            }
+                        }
+                        if (anyFalse)
+                        {
+                            break;
+                        }
+                        else
+                        {
+                            subCheckResultsResults.Add(tempCheck);
+                            methodParameters.RemoveAt(x);
+                            noneCorrect = false;
+                            break;
+                        }
+                    }
+                }
+
+                // reached the last iteration none were correct
+                if (x == methodParameters.Count() - 1 && noneCorrect)
+                {
+                    subCheckResultsResults.Add(tempCheck);
+                }
+            }
+        }
+
+        // TODO feedback message
+        return new NodeCheckResult
+        {
+            ChildrenCheckResults = subCheckResultsResults, 
+            FeedbackMessage = String.Empty, 
```

Comment:
Maak van de FeedbackMessage een message van het volgende format: $"Found parameters for following node: {node}."